src/seedsigner/models/psbt_parser.py

```diff
@@ -390,3 +415,58 @@ def verify_multisig_output(self, descriptor: Descriptor, change_num: int) -> boo
         is_owner = descriptor.owns(output)
         # print(f"{self.psbt.tx.vout[i].script_pubkey.address()} | {output.value} | {is_owner}")
         return is_owner
+
+
+    def fill_zero_fingerprints(self):
+        """Fix for zeros in fingerprint that happen when user imports the wallet
+        with XPUB only (without derivation path)
+        """
+        if not self.root:
+            return 0
+        
+        filled = 0
+
+        for inp in self.psbt.inputs:
+            filled += self._fill_zero_fingerprint_scope(inp)
+
+        for out in self.psbt.outputs:
+            filled += self._fill_zero_fingerprint_scope(out)
+
+        if filled > 0:
+            logger.info(f"Filled {filled} zero fingerprints with correct fingerprint")
+
+        return filled
+
+
+    def _fill_zero_fingerprint_scope(self, scope: Union[InputScope, OutputScope]):
+        """Helper function to fill zero fingerprints in a scope (input/output)"""
```

Comment:
Can you add a comment either here or additional comments in the function body to explain how this works?

The crucial check is on line 452.

e.g.
"If the blah blah match, then we can be sure that the input/output is owned by the signing key and we can go ahead and fill in the fingerprint with the signing seed's fingerprint."